### DISCUSSION TITLE
Getting all of front matter 'attributes'

### DISCUSSION THREAD

**Discussion by cekvenich2** [11ty/eleventy#2124] [Q&A]
Hi, I'm getting it like this starting at line 10:

https://gitlab.com/cekvenich2/fe/-/blob/main/.eleventy.js

It works. Is there a better way?

(Also, I love 11ty and I love how community was helpful to me when I started)

**Comment by pdehaan** [comment]
Not sure I understand what the code is doing:
```js
eleventyConfig.addCollection("blogs", function (collection) {
  let col = [];

  collection.getAll().filter(function (item) {
    if (!item.data.page.url.includes("/posts")) return false;

    for (const tag in item.data) {
      if (tag == "page") continue;
      if (tag == "pkg") continue;
      if (tag == "eleventy") continue;
      if (tag == "collections") continue;
      item.data.page[tag] = item.data[tag]; // upgrade
    }
    console.log(item.data.page); // upgraded
    col.push(item.data.page);
    return true;
  });

  return col;
});
```

So you're getting all pages from Eleventy, via`.getAll()`, then exiting early if the page is in /posts/* folder, then looping over all front matter via `item.data`. And then ignoring the data property if it is `page`, `pkg`, `eleventy`, or `collections`. Otherwise, copy the current key/value into the current page's `page` scope.

You could probably reduce a few lines by doing something like this:
```js
if (["page", "pkg", "eleventy", "collections"].includes(tag))
	continue;
```

But I guess I'm just curious what you're trying to achieve. Are you just trying to find all blog posts in the /blog directory? Or posts with specific tags?

**Comment by cekvenich2** [comment]
What I do is look for posts/blogs: and get all the front matter 'tag'. It
works. The purpose is to have a list of blogs and 'meta' data about it, so
I can show the list of blog posts.

But it seems to be a common need, should be easier? If there is no easier
way, it's cool.
